src/cloud_provider/service.rs

```diff
@@ -470,6 +471,68 @@ where
     Ok(())
 }
 
+pub fn scale_down_database(
+    target: &DeploymentTarget,
+    service: &impl Database,
+    replicas_count: usize,
+) -> Result<(), EngineError> {
+    let (kubernetes, environment) = match target {
+        DeploymentTarget::ManagedServices(_, _) => {
+            info!("Doing nothing for pause database as it is a managed service");
+            return Ok(());
+        }
+        DeploymentTarget::SelfHosted(k, env) => (*k, *env),
+    };
+
+    let scaledown_ret = kubectl_exec_scale_replicas(
+        kubernetes.config_file_path()?,
+        kubernetes.cloud_provider().credentials_environment_variables(),
+        environment.namespace(),
+        Statefulset,
+        service.sanitized_name().as_str(),
+        replicas_count as u32,
```

Comment:
Even if not likely to happen, maybe worth being more cautious one this conversion?
What about using `usize::try_from(replicas_count).unwrap();`?